Wrappers/Python/cil/processors/TransmissionAbsorptionConverter.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+#  CCP in Tomographic Imaging (CCPi) Core Imaging Library (CIL).
+
+#   Copyright 2017 UKRI-STFC
+#   Copyright 2017 University of Manchester
+
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from cil.framework import DataProcessor, AcquisitionData, ImageData, DataContainer
+import warnings
+import numpy
+
+
+class TransmissionAbsorptionConverter(DataProcessor):
+
+    '''
+    Processor to convert from transmission measurements to absorption
+    based on the Beer-Lambert law
+    
+    Processor first divides by white_level (default=1) and then take negative logarithm. 
+    Elements below threshold (after division by white_level) are set to value.
+    '''
+
+    def __init__(self,
+                 threshold = 1e-3,
```

Comment:
This seems like a fairly high default threshold. I would probably set the default to zero and have sensible error handling if log(0) is an issue with a hint to the user to set the threshold.

I also wonder if we could threshold on maximum attenuation which is maybe more intuitive to the user than thresholding on minimum intensity. Or take in both/either?

Reply:
Hmm, I think threshold on minimum intensity is more intuitive, as threshold on maximum attenuation requires some prior knowledge on physical properties of an object

Reply:
I'm not sure I agree, For an intensity threshold you still need to know what the lowest intensity you expect is and now that's dependant on the object attenuation coefficient, path lengths and I0. To just avoid infinite attenuations you'll threshold at the smallest unit you can (so if data is stored as unsigned ints presumably 1/65535). But for any more meaningful threshold I think you'd use attenuation, even if you calculate it and convert it to intensity.

I'm happy leaving it as a threshold on intensity anyway, but we could change `threshold` to a more specific name i.e. `intensity_threshold` `min_intensity`, `threshold_intensity` to allow us to extend it in the future, or we at least make it clear in the doc string this is the minimum intensity value and whether it's applied before or after normalisation.

Reply:
I chenged to `min_intensity` and updated docstring

Reply:
perfect :)